tachyon/zk/plonk/halo2/proof.h

```diff
@@ -125,24 +107,90 @@ struct Proof {
         gamma,
     };
   }
+};
+
+template <typename F, typename C>
+struct Halo2Proof : public Proof<F, C> {
```

Comment:
What do you think about moving Halo2Proof and LogUpProof to their respective folders?

Reply:
wouldn't it better to have all the arguments of the proof in a single file for readability? Or we should see multiple files for one type of proof. Also if there will be more various types of Halo2 proof, then it makes sense. But I think there won't be more types of Halo2 proof ig.